mrmustard/physics/ansatze.py

```diff
@@ -787,6 +787,84 @@ def __call__(self, z: Batch[Vector]) -> Scalar:
             )  # (b_arg)
         return val
 
+    def call_none(self, z: Batch[Vector]) -> DiffOpPolyExpAnsatz:
+        r"""
+        Updates the ansatz from calling the ansatz on some wires, while leaving the rest as None.
+        Note that the batch of the triple and argument in this method is handled parwise, unlike the regular call where the batch over the triple is a superposition.
+        Args:
+            z: slice in C^n where the function is evaluated, while unevaluated along other axes of the space.
+
+        Returns:
+            A new ansatz, which is a "slice" of the old one.
+        """
+
+        def call_none_single(Ai, bi, ci, zi):
```

Comment:
Can we move this outside of the scope of ``call_none``? You could add it as a private method of this class with an underscore in front.